discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,376 @@
+        )
+
+        logger.info('deleted cluster deployment %s', self.ref)
+
+    def status(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT
+    ) -> dict:
+
+        # 'status' key is created only after resource is processed by the
+        # controller in assisted-service. It might take a few seconds before
+        # appears.
+        def _attempt_to_get_status() -> dict:
+            return self.get()['status']
+
+        return waiting.wait(
+            _attempt_to_get_status,
+            sleep_seconds=0.5,
+            timeout_seconds=timeout,
+            waiting_for=f'cluster {self.ref} status',
+            expected_exceptions=KeyError
+        )
+
+    def state(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> Tuple[str, str]:
+
+        state, state_info = None, None
+        for condition in self.status(timeout).get('conditions', []):
+            reason = condition.get('reason')
+
+            if reason == 'AgentPlatformState':
+                state = condition.get('message')
+
+            # todo: fix typo after it will be fixed in assisted-service
+            #  (Agnet -> Agent)
+            elif reason == 'AgnetPlatformStateInfo':
+                state_info = condition.get('message')
+
+            if state and state_info:
+                break
+
+        return state, state_info
+
+    def wait_for_state(
+            self,
+            required_state: str,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> None:
+
+        required_state = required_state.lower()
+
+        def _has_required_state() -> bool:
+            state, _ = self.state(0.5)
+            return state.lower() == required_state
+
+        waiting.wait(
+            _has_required_state,
+            timeout_seconds=timeout,
+            waiting_for=f'cluster {self.ref} state to be {required_state}',
+            expected_exceptions=waiting.exceptions.TimeoutExpired
+        )
+
+
+def deploy_default_secret(
+        name: str,
+        ignore_conflict: bool = True,
+        pull_secret: Optional[str] = None
+) -> Secret:
+
+    if pull_secret is None:
+        pull_secret = os.environ.get('PULL_SECRET', '')
+    secret = Secret(name)
+    try:
+        secret.create(pull_secret)
+    except ApiException as e:
+        if e.reason != 'Conflict' or not ignore_conflict:
+            raise
+    return secret
+
+
+def deploy_default_cluster_deployment(
+        name: str,
+        ignore_conflict: bool = True,
+        base_domain: str = env_variables['base_domain'],
+        platform_params: Optional[dict] = None,
+        install_strategy_params: Optional[dict] = None,
+        secret: Optional[Secret] = None,
+        **kwargs
+) -> ClusterDeployment:
+
+    platform = Platform(**platform_params or {})
+    install_strategy = InstallStrategy(**install_strategy_params or {})
```

Comment:
Use `dict()` as a default var to avoid unnecessary `None`
```python
        platform_params: dict = dict(),
        install_strategy_params: dict = dict(),
        secret: Optional[Secret] = None,
        **kwargs
) -> ClusterDeployment:

    platform = Platform(**platform_params)
    install_strategy = InstallStrategy(**install_strategy_params)
```

Reply:
No, it's a bad practice to use mutable objects as default values. 
Take for example this class:
```
class Test(object):

    def __init__(self, key, value, dct = dict()):
        self.dct = dct
        self.dct[key] = value


t1 = Test('a', 'b')
t2 = Test('c', 'd')

print(t1.dct)
print(t2.dct)
```
You would expect that t1.dct will be `{'a': 'b'} `and t2.dct will be `{'b': 'c'}`, right?
But since I used a mutable object as default, I will get:
```
{'a': 'b', 'c': 'd'}
{'a': 'b', 'c': 'd'}
```
That is why I need to use None.